subgraphs/aave-v2/src/mappings/lendingPoolAddressProvider.ts

```diff
@@ -0,0 +1,109 @@
+  log.info('HANDLING PRICE ORACLE UPDATE TO ' + event.params.newAddress.toHexString(), []);
+  const lendingProtocol = fetchProtocolEntity(protocolAddress);
+  lendingProtocol.protocolPriceOracle = event.params.newAddress.toHexString();
+  lendingProtocol.save();
+}
+
+export function handleLendingPoolUpdated (event: LendingPoolUpdated): void {
+  const context = initiateContext(event.address);
+  startIndexingLendingPool(event.params.newAddress, context);
+}
+
+export function handleLendingPoolConfiguratorUpdated (event: LendingPoolConfiguratorUpdated): void {
+  const context = initiateContext(event.address);
+  startIndexingLendingPoolConfigurator(event.params.newAddress, context);
+}
+
+export function startIndexingLendingPool (poolAddress: Address, context: DataSourceContext): void {
+  // Create a template for an implementation of a Lending Pool/Market
+  // This indexes for events which users act upon a lending pool within the lendingPool.ts mapping script
+  log.info('START INDEXING LENDING POOL', []);
+  LendingPoolTemplate.createWithContext(poolAddress, context);
+}
+
+export function startIndexingLendingPoolConfigurator (configurator: Address, context: DataSourceContext): void {
+  // Create a template for an implementation of a Lending Pool Configurator
+  // This indexes for events within the lendingPoolConfigurator.ts mapping script
+  log.info('START INDEXING LENDING POOL CONFIG', []);
+  LendingPoolConfiguratorTemplate.createWithContext(configurator, context);
+}
+
+function initiateContext (addrProvider: Address): DataSourceContext {
+  // Add Lending Pool address, price oracle contract address, and protocol id to the context for general accessibility
+  const contract = AddressProviderContract.bind(addrProvider);
+  log.info('AddrProvContract: ' + addrProvider.toHexString(), []);
+  // Get the lending pool
+  const trylendingPool = contract.try_getLendingPool();
+  let lendingPool = ''
+  if (!trylendingPool.reverted) {
+    lendingPool = trylendingPool.value.toHexString();
+    log.info('initiateContext LP:' + lendingPool, []);
+  }
+
+  // Initialize the protocol entity
+  const lendingProtocol = fetchProtocolEntity(protocolAddress);
+  // Get the Address Provider Contract's Price Oracle
+  const tryPriceOracle = contract.try_getPriceOracle();
+  if (!tryPriceOracle.reverted) {
+    lendingProtocol.protocolPriceOracle = tryPriceOracle.value.toHexString();
+    log.info('initiateContext priceOracle: ' + lendingProtocol.protocolPriceOracle, []);
+  } else {
+    lendingProtocol.protocolPriceOracle = priceOracleDefault;
+    log.error('FAILED TO GET ORACLE - REVERTED TO DEFAULT HARD-CODED AT ' + lendingProtocol.protocolPriceOracle, ['']);
+  }
+  lendingProtocol.save();
+  log.info('CREATING CONTEXT ' + lendingPool + '-----' + lendingProtocol.id, []);
+  const context = new DataSourceContext();
+  context.setString('lendingPool', lendingPool);
+  context.setString('protocolId', lendingProtocol.id);
```

Comment:
Since this doesn't change across the entire subgraph, why not just use the constant instead of passing context around?

Reply:
The reason that I used context is because the addresses are dynamically set by events on the implementation contract. From my understanding, throughout the lifespan of V2 the Lending Pool and Price Oracle proxy contract addresses do not change. However, for the sake of setting things dynamically I set it to context to be both changeable and accessible to its current state. Would it be fine in this case to just have proxy addresses hardcoded rather than dynamically set them? Im still not completely clear on what good practice is regarding dynamically setting addresses vs hardcoding constants for proxy contract addresses.

Reply:
The reason that I used context is because the addresses are dynamically set by events on the implementation contract. From my understanding, throughout the lifespan of V2 the Lending Pool and Price Oracle proxy contract addresses do not change. However, for the sake of setting things dynamically I set it to context to be both changeable and accessible to its current state. Would it be fine in this case to just have proxy addresses hardcoded rather than dynamically set them? Im still not completely clear on what good practice is regarding dynamically setting addresses vs hardcoding constants for proxy contract addresses.